libs/vkd3d-shader/spirv.c

```diff
@@ -2389,7 +2390,7 @@ static const struct vkd3d_shader_resource_binding *vkd3d_dxbc_compiler_get_resou
     {
         const struct vkd3d_shader_resource_binding *current = &shader_interface->bindings[i];
 
-        if (!(current->flags & binding_flag))
+        if ((current->flags & binding_flags) != binding_flags)
```

Comment:
Hm, how would this match against "non-SSBO" case? E.g. if we have SHADER_BINDING_FLAG_BUFFER | RAW_SSBO and we query for plain FLAG_BUFFER, we would get a false match I think?

Reply:
Yeh, this should probably test for equality rather than ANDing flags.

But this aspect of the dxbc compiler is so broken that I'm not really sure, needs testing. It currently works because we always define raw SSBO bindings after any others.

Reply:
Could leave it and add a comment about it at least?

Reply:
Checking for equality does not work, everything just crashes and burns. Not sure why.

Can add a comment.

Reply:
Ah, actually - I guess we should just ignore certain flags, such as `VKD3D_SHADER_BINDING_FLAG_BINDLESS`, when doing this check.

Reply:
Fixed.